src/test/java/org/broadinstitute/barclay/argparser/LegacyCommandLineArgumentParserTest.java

```diff
@@ -608,52 +608,151 @@ public void testAddToDefaultValuesListOption() {
 
-        overrideClp.parseArguments(System.err, new String[]{"STRING3=String3Supplied"});
+        String[] args = {"REFERENCE_SEQUENCE=ref.fasta"}; // with reference provided
+        Assert.assertTrue(clp.parseArguments(System.err, args));
+        Assert.assertEquals(o.referenceSequence.getReferenceFile().getName(), "ref.fasta");
+    }
+
+    // Tool with required reference, no reference provided (parseArguments issues error message; returns false)
+    @Test
+    public void testArgumentCollectionNoReferenceProvidedForRequired() {
+        final ToolWithRequiredReference o = new ToolWithRequiredReference();
+        final LegacyCommandLineArgumentParser clp = new LegacyCommandLineArgumentParser(o);
+
+        String[] args = {}; // no reference provided
+        Assert.assertFalse(clp.parseArguments(System.err, args));
+    }
 
-        final OverridePropagation propsSet = (OverridePropagation) overrideClp.getCallerOptions();
-        Assert.assertTrue(propsSet.STRING3.equals("String3Supplied"));
-        Assert.assertTrue(((StaticParent) propsSet).STRING3.equals("String3Supplied"));
+    // Tool with required reference, with reference provided
+    @Test
+    public void testArgumentCollectionWithReferenceProvidedForRequired(){
+        final ToolWithRequiredReference o = new ToolWithRequiredReference();
+        final LegacyCommandLineArgumentParser clp = new LegacyCommandLineArgumentParser(o);
+
+        String[] args = {"REFERENCE_SEQUENCE=ref.fasta"}; // with reference provided
+        Assert.assertTrue(clp.parseArguments(System.err, args));
+        Assert.assertEquals(o.referenceSequence.getReferenceFile().getName(), "ref.fasta");
     }
 
+    //////////////////////////////
+    // Nested argument collections
+
+    class ArgumentCollectionContainerInner {
+        @ArgumentCollection
+        ArgsCollection innerContainerArg = new ArgsCollection();
+    }
+
+    class ArgumentCollectionContainerOuter {
+        @Argument(fullName = "outerContainerArg")
+        public int outerContainerArg;
+
+        @ArgumentCollection
+        ArgumentCollectionContainerInner argInner = new ArgumentCollectionContainerInner();
+    }
+
+    class ToolWithNestedArgumentCollection  {
+        @ArgumentCollection
+        public ArgumentCollectionContainerOuter argOuter =  new ArgumentCollectionContainerOuter();
+    }
+
+    @Test
+    public void testArgumentCollectionNested() {
+        final ToolWithNestedArgumentCollection o = new ToolWithNestedArgumentCollection();
+        final LegacyCommandLineArgumentParser clp = new LegacyCommandLineArgumentParser(o);
+
+        String[] args = {"outerContainerArg=17", "arg1=92"};
+        Assert.assertTrue(clp.parseArguments(System.err, args));
+        Assert.assertEquals(o.argOuter.outerContainerArg, 17);
+        Assert.assertEquals(o.argOuter.argInner.innerContainerArg.Arg1, 92);
+    }
+
+    private List<String> makeList(final String... list) {
```

Comment:
Might as well make this static.  Not that it's particularly important in a test class...